course/courseshowitems.php

```diff
@@ -1925,10 +1944,10 @@ function quickview($items,$parent,$showdates=false,$showlinks=true) {
 			   }
 			   if ($showlinks) {
 				   echo '<span class="links">';
-				    echo " <a href=\"addquestions.php?aid=$typeid&cid=$cid\">", _('Questions'), "</a> | <a href=\"addassessment.php?id=$typeid&cid=$cid\">", _('Settings'), "</a> | \n";
-				   echo "<a href=\"deleteassessment.php?id=$typeid&block=$parent&cid=$cid&remove=ask\">", _('Delete'), "</a>\n";
+				   echo " <a href=\"addquestions.php?aid=" . Sanitize::onlyInt($typeid) . "&cid=$cid\">", _('Questions'), "</a> | <a href=\"addassessment.php?id=$typeid&cid=$cid\">", _('Settings'), "</a> | \n";

```

Comment:
`$typeid` is only being sanitized in some places it's used.  Though being an int field from the DB, sanitizing at all is pretty unnecessary.